File: Notification-Service/src/main/java/byrd/riley/notificationservice/app/MQConfiguration.java
package byrd.riley.notificationservice.app;

import byrd.riley.notificationservice.model.Emailer;
import byrd.riley.notificationservice.model.QueueReceiver;
import byrd.riley.notificationservice.model.Texter;
import org.springframework.amqp.core.Queue;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.JavaMailSenderImpl;

import byrd.riley.notificationservice.model.TwilioTexter;

@Configuration
public class MQConfiguration {

	@Bean
	public Queue queue() {
		return new Queue("task_queue");
	}
	
	@Bean
	public Texter texter() {
		return new TwilioTexter();
	}
	
	@Bean
	public JavaMailSender javaMailSender() {
		return new JavaMailSenderImpl();
	}
	
	@Bean
	public Emailer emailer() {
		return new Emailer(javaMailSender());
	}
	
	@Bean
	public QueueReceiver queueReceiver() {
		return new QueueReceiver(texter(), emailer());
	}
}

File: Notification-Service-Kotlin/src/main/kotlin/byrd/riley/notificationservicekotlin/app/MQConfiguration.kt
package byrd.riley.notificationservicekotlin.app

import byrd.riley.notificationservicekotlin.model.Emailer
import byrd.riley.notificationservicekotlin.model.QueueReceiver
import byrd.riley.notificationservicekotlin.model.Texter
import byrd.riley.notificationservicekotlin.model.TwilioTexter
import org.springframework.amqp.core.Queue
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.mail.javamail.JavaMailSenderImpl

@Configuration
class MQConfiguration {

    @Bean
    fun queue(): Queue = Queue("task_queue")

    @Bean
    fun texter(): Texter = TwilioTexter()

    @Bean
    fun javaMailSender(): JavaMailSender = JavaMailSenderImpl()

    @Bean
    fun emailer(): Emailer = Emailer(javaMailSender())

    @Bean
    fun queueReceiver(): QueueReceiver = QueueReceiver(texter(), emailer())
}